### PR TITLE
Improvements to python course and intermediate

### PR DESCRIPTION
**Pull request by louisdewar**

### CODE REVIEW

File: courses/python/exercises.yml [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+
+      # Prints '3' (a = 0, b = 1, ...)
+      print(alphabet.index('d'))
+
+      # Prints 4 (% returns the remainder of the division - look into modular arithmetic)
+      print(30 % len(alphabet))
+      ```
+
+      ## Test it
+      Try running `caesar_shift('cat', 2)`, it should equal `ECV`.
```
Comment (marcelloromani) [inline courses/python/exercises.yml]: Either `CAT` or `ecv` ;-)

File: courses/python/exercises.yml [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+  start_closed: true
+
+tutorials:
+  - subtitle: Introduction
+    start_closed: false
+    content: |
+      Before you start this course you should have a basic understanding of python.
+
+      If you want to recap see the python tutorials on the website
+      (particularily `introduction` and `intermediate`).
```
[the remark this answers is not in the document]
Comment (louisdewar) [reply courses/python/exercises.yml]: I'll add those links

File: courses/python/exercises.yml (hunk 1 above)
Comment (louisdewar) [reply courses/python/exercises.yml]: good spot

File: courses/python/exercises.yml [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+      and it returns the factorial of the number.
+
+      ## Test it
+      To make sure it works check it returns the correct results to these
+      (use `print(factorial(n))` where n is the number):
+      - `factorial(3) = 6`
+      - `factorial(5) = 120`
+
+      ## Research
+      What happens if you pass in `0!` to your program?
```
[the remark this answers is not in the document]
Comment (louisdewar) [reply courses/python/exercises.yml]: fixed

**Comment by louisdewar** [comment]
@marcelloromani Thanks for that, obviously still need to work on the intermediate tutorial (see the TODO: comment at the bottom).

**Comment by louisdewar** [comment]
I think I'm going to merge this as it is since most of the content here is definitely acceptable, obviously the intermediate tutorial isn't finished but we can always come back to it later.

**Pull request merged**